src/cr/cube/matrix/assembler.py

```diff
@@ -1127,6 +1134,26 @@ def _subtotal_values(self):
         return measure_subtotal_rows[:, self._column_idx]
 
 
+class _SortRowsByDerivedColumnHelper(_SortRowsByBaseColumnHelper):
+    """Orders elements by the values of an opposing derived element vector.
+
+    This would be like "order rows in descending order by value of 'Coke' or 'Diet Coke'
+    array column. An opposing-element ordering is only available on a matrix, because only
+    a matrix dimension has an opposing dimension.
+
+    In user-facing language, these "derived columns" are called insertions, the difference
+    being that these insertions must be calculated by zz9.
+    """
+
+    @lazyproperty
+    def _column_idx(self):
+        """int index of column whose values the sort is based on."""
+        # --- MR insertions are called derived elements by cube
+        derived_element_ids = self._columns_dimension.derived_element_ids
+        insertion_id = self._order_spec.insertion_id
+        return derived_element_ids.index(insertion_id)
```

Comment:
I'm not sure I get this concept 100%. I'd love a ☕  session together, so that you can take me through this, at your convenience.

Reply:
@gergness are we sure that `insertion_id` matches the `element_id` in all cases? I'm not at all sure that it does.

Such an insertion can arise in the variable-definition, in which case maybe we can depend on those matching, I don't know. But they can also arise in an analysis-definition and I'm not sure how they get numbered there. However they do get numbered there can vary by the client. There's no centralized "authority" that assigns these uniformly.

I wouldn't be surprised if this passes the test by accident of normal sequential `1, 2, 3, ...` ordering or something. I'd say we need to be confident of the contracts we're counting on to identify this inserted element and to document them in the comments here for future folks who come through and need to reconstruct how it works (and possibly reason about why it's now broken :).